### PR TITLE
remove checks for Ice version < 3.6

### PR DESCRIPTION
**Pull request by mtbc**
# What this PR does

Removes code paths that apply only to Ice < 3.6.

# Testing this PR

CI should remain green. Also check that can still use 5.4 devspace web client and Insight client to connect.

# Related reading

http://www.openmicroscopy.org/site/support/omero5/sysadmins/version-requirements.html#ice

### CODE REVIEW

**Comment by jburel** [comment]
This might be better to be included post 5.4.0-m1

File: components/blitz/src/omero/client.java [hunk 1]
```diff
@@ -350,11 +346,8 @@ private void init(Ice.InitializationData id) {
         optionallySetProperty(id, "omero.block_size", Integer
             .toString(omero.constants.DEFAULTBLOCKSIZE.value));
 
-        // Set the default encoding if this is Ice 3.5 or later
-        // and none is set.
-        if (Ice.Util.intVersion() >= 30500) {
-            optionallySetProperty(id, "Ice.Default.EncodingVersion", "1.0");
-        }
+        // Set the default encoding if none is set.
+        optionallySetProperty(id, "Ice.Default.EncodingVersion", "1.0");
```
Comment (rleigh-codelibre) [inline components/blitz/src/omero/client.java:350]: With Ice 3.6 and above being the only supported versions, we could also look at increasing the encoding version here as well.  The same applies to other instances of the encoding version in this PR and the codebase in general.  May *possibly* have some impact on shares, but I'm not certain of that.

**Comment by jburel** [comment]
omero.properties still has 3.5
As indicated in standup, other sections will need to be reviewed at the same time
It will be preferable to have a card with a list of actions e.g. ci, repo to check so it is "less random" before opening a PR that will break the daily build

**Comment by jburel** [comment]
and we might need similar checks with 3.7 so this could go in a "3.7"support effort

**Comment by mtbc** [comment]
I don't believe that this PR would break the daily build unless it's relying on deploying the Ice 3.5 artifacts?

As discussed in person, better to make the various Ice changes all in one PR so some are not forgotten. I don't know what all those are and am not planning to investigate right now so closing this PR.

**Pull request closed without merging**